File: 백준/Silver/33554. Grand Opening/Grand Opening.c
#include <stdio.h>
#include <string.h>

#define MAX_N 1000
#define MAX_W 100
#define MAX_H 100

char rocks[MAX_N][MAX_H][MAX_W + 1];
char key[MAX_H][MAX_W + 1];

int main() {
    int n, w, h;
    scanf("%d %d %d", &n, &w, &h);
    
    for (int r = 0; r < n; r++) {
        for (int i = 0; i < h; i++) {
            scanf("%s", rocks[r][i]);
        }
    }
    
    for (int i = 0; i < h; i++) {
        scanf("%s", key[i]);
    }
    
    int ans = 0;
    
    for (int r = 0; r < n; r++) {
        int valid = 1;
        for (int l = 1; l <= w; l++) {
            for (int i = 0; i < h && valid; i++) {
                for (int j = 0; j < l; j++) {
                    if (rocks[r][i][j] == '#' && key[i][w + j - l] == '#') {
                        valid = 0;
                        break;
                    }
                }
            }
            if (!valid) break;
        }
        if (valid) ans++;
    }
    
    printf("%d\n", ans);
    return 0;
}

File: 백준/Silver/33554. Grand Opening/Grand Opening.py
import sys

input = sys.stdin.read
data = input().split()

n, w, h = map(int, data[:3])
offset = 3
rocks = [data[offset + i * h : offset + (i + 1) * h] for i in range(n)]
key = data[offset + n * h : offset + (n + 1) * h]

ans = 0
for rock in rocks:
    valid = True
    for l in range(1, w + 1):
        found = False
        for i in range(h):
            for j in range(l):
                if rock[i][j] == '#' and key[i][w + j - l] == '#':
                    found = True
                    break
            if found:
                break
        if found:
            valid = False
            break
    if valid:
        ans += 1

print(ans)